tools/check_format.py

```diff
@@ -110,15 +111,31 @@ def checkFilePath(file_path):
     return
   checkNamespace(file_path)
   checkProtobufExternalDeps(file_path)
-  command = ("%s %s | diff -q %s - > /dev/null" % (HEADER_ORDER_PATH, file_path,
+  command = ("%s %s | diff %s -" % (HEADER_ORDER_PATH, file_path,
                                                    file_path))
-  if os.system(command) != 0:
-    printError("header_order.py check failed for file: %s" % (file_path))
-  command = ("%s %s | diff -q %s - > /dev/null" % (CLANG_FORMAT_PATH, file_path,
-                                                   file_path))
-  if os.system(command) != 0:
-    printError("clang-format check failed for file: %s" % (file_path))
+  executeCommand(command, "header_order.py check failed", file_path)
 
+  command = ("%s %s | diff %s -" % (CLANG_FORMAT_PATH, file_path,
+                                                   file_path))
+  executeCommand(command, "clang-format check failed", file_path)
+
+# Example target outputs are:
+#   - "26,27c26"
+#   - "12,13d13"
+#   - "7a8,9"
+def executeCommand(command, error_message, file_path,
+        regex=re.compile(r"^(\d+)[a|c|d]?\d*(?:,\d+[a|c|d]?\d*)?$")):
+  try:
+    subprocess.check_output(command, shell=True, stderr=subprocess.STDOUT)
+  except subprocess.CalledProcessError as e:
+      if (e.returncode != 0 and e.returncode != 1):
+          print "ERROR: something went wrong while executing: %s" % e.cmd
+          sys.exit(1)
+      # In case we can't find any line numbers, call printError at first.
+      printError(error_message + " for file: %s" % (file_path))
```

Comment:
consistent string formatting please
i.e. `"%s for file: %s" % (error_message, file_path)`